digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataExporter/AssetReportCsvExporter.scala

```diff
@@ -0,0 +1,264 @@
+        withDynSession {
+          val municipalityInfo = municipalityDao.getMunicipalityIdByName(municipalityName)
+          assetReporterDAO.getTotalTrafficSignNewLaw(municipalityInfo.head.id)
+        }
+      }
+      else{
+        val municipalityInfo = municipalityDao.getMunicipalityIdByName(municipalityName)
+        assetReporterDAO.getTotalTrafficSignNewLaw(municipalityInfo.head.id)
+      }
+    }
+
+
+    val output = if (exportAssetReports.isEmpty) {
+        List()
+      }
+      else {
+        val allInfoGroupedByMunicipality = exportAssetReports.groupBy(_.municipalityName)
+
+        val result = allInfoGroupedByMunicipality.map { case (municipality, records) =>
+          val line = municipality + ";\r\n"
+
+          val rows = records.map { elem =>
+                val (totalPoints, isLinear) = if (elem.assetGeomType.trim == "linear") ("", "Kyllä")
+                                              else (elem.totalAssets, "")
+
+                val operatorDate = extractDateToString(elem.operatorModifiedDate)
+                val municipalityDate = extractDateToString(elem.municipalityModifiedDate)
+
+                val baseRow = s";${elem.assetNameFI};$totalPoints;$isLinear;$operatorDate;$municipalityDate;${elem.operatorUser};${elem.municipalityUser}"
+
+                if (elem.assetTypeId == TrafficSigns.typeId) {
+                  val totalNewLawTrafficSigns = getTotalNewLawTrafficSigns(municipality)
+                  baseRow.concat(s";$totalNewLawTrafficSigns")
+                }
+                else
+                  baseRow
+                }
+
+          line + rows.mkString("\r\n")
+        }
+
+      result.toList
+    }
+
+    output.mkString("\r\n")
+  }
+
+
+  def decodeAssetsToProcess(assetTypesList: List[Int]): List[Int] = {
+
+    val ( pointAssets, linearAssets ) = ( assetTypesList.contains(PointAssets.id), assetTypesList.contains(LinearAssets.id) )
+
+    if ( assetTypesList.contains(AllAssets.id) ) {
+      AssetTypeInfo.values.map(_.typeId).toList
+    }
+    else if ( pointAssets || linearAssets ) {
+
+      val (linearAssetType, pointAssetType) = AssetTypeInfo.values.partition( _.geometryType == "linear" )
+
+      val allPoints = pointAssetType.map(_.typeId)
+      val allLinears = linearAssetType.map(_.typeId)
+
+      (pointAssets, linearAssets) match {
+        case (true, false) =>
+          (allPoints ++ assetTypesList).filterNot(_ == PointAssets.id).toList
+
+        case (false, true) =>
+          (allLinears ++ assetTypesList).filterNot(_ == LinearAssets.id).toList
+
+        case (true, true) =>
+          (allPoints ++ allLinears).filterNot(id => id == PointAssets.id || id == LinearAssets.id).toList
+
+        case _ =>
+          List()
+      }
+
+    }
+    else
+      assetTypesList
+  }
+
+
+  def decodeMunicipalitiesToProcess(municipalities: List[Int]): List[Int] = {
+    if (municipalities.isEmpty || municipalities.contains(1000)) { /* 1000 = All municipalities */
```

Comment:
Is possible to send a empty list of municipalities?

Reply:
well...it is bug :D

Reply:
Validate if the Municipalities list is not empty when receive it and then:

```Scala
  def decodeMunicipalitiesToProcess(municipalities: List[Int]): List[Int] = {
    if (municipalities.contains(1000)) { /* 1000 = All municipalities */
      withDynSession {
        municipalityDao.getMunicipalitiesInfo.map(_._1).toList
      }
    }
    else
      municipalities
  }

Reply:
Done